test/e2e/util.go

```diff
@@ -679,7 +680,13 @@ func waitForNamespacesDeleted(c *client.Client, namespaces []string, timeout tim
 }
 
 func waitForServiceAccountInNamespace(c *client.Client, ns, serviceAccountName string, timeout time.Duration) error {
-	Logf("Waiting up to %v for service account %s to be provisioned in ns %s", timeout, serviceAccountName, ns)
+	w, err := c.ServiceAccounts(ns).Watch(api.SingleObject(api.ObjectMeta{Name: serviceAccountName}))
```

Comment:
After further thought, the watch with no RV case sends the current object and then changes, so as long as the condition is level triggered, this should be OK. I think?